zk-tutorials/proof-system/groth16/theory_to_practice.md

```diff
@@ -5,10 +5,10 @@ The [Groth16](https://eprint.iacr.org/2016/260.pdf) algorithm is a non-interacti
 Groth16 solves the problem of how to perform zero-knowledge proofs for complex computations. In zero-knowledge proof, the prover wishes to prove a statement about private inputs to the verifier, without revealing any information about the inputs themselves. This input is usually referred to as witness in ZkSnark. It is a solution satisfying certain conditions. The Groth16 proof system allows the prover to create a proof that they know a solution without revealing the actual solution to the verifier. As we previously introduced the basic concept of ZKSNARK(zero-knowledge succinct non-interactive argument of knowledge), it is a specific algorithm implementation of ZKSNARK.
 
 ### Features:
-- **Non-interactive**: Groth16 is a non-interactive proof system, meaning that interaction is not needed between the prover and the verifier.
-- **Short proofs**: Groth16 proofs are short, with a constant size of a few hundred bytes, which remains independent of the complexity of the proof.
-- **Fast verification**: Groth16 proofs can be verified efficiently in polynomial time.
-- **Trusted setup**: Groth16 requires a trusted setup phase for generating common parameters.
+- **Non-interactive**: Groth16 is a non-interactive proof system, meaning that only one communication is needed between the prover and the verifier.
+- **Short proofs**: Groth16 proofs are short, with a size of only O(1), independent of the complexity of the proof.
+- **Fast verification**: Groth16 proofs can be verified in polynomial time.
+- **Trusted setup**: Groth16 requires a trusted setup phase for generating common parameters.[skalman](# "arguably this is not a feature, this is a limitation, maybe you should move it to limitation section")
```

Comment:
Maybe we can leave it here? Here I'm describing are the features of groth16, not the advantages of groth16.